zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -427,8 +435,8 @@ private BatchProcessingStepResult collectBatchProcessingStepResult(
                         context.getPartitionId(),
                         entry.recordValue(),
                         entry.recordMetadata()));
-                toWriteEntry = LogAppendEntry.ofProcessed(entry);
               }
+              toWriteEntry = LogAppendEntry.ofProcessed(entry, lastProcessedCommandIndex);
```

Comment:
:x: This might work right now but I think it's conceptually wrong. Here, we get an `entry` that might already have some source index set and wrap it to mark it as processed. With your changes, we now overwrite the source index too. But in principle, the engine is free to set its own source index and we should not overwrite it just like that.

I would suggest sticking to what I suggested initially: making the `currentSourceIndex` a property of the `BufferedProcessingResultBuilder`. We update it during batch processing so that the result builder can use this index instead of `-1` if users of that result builder don't provide their own source index.

Does that make sense so far?

Reply:
Reverted the changes to `LogAppendEntry`.